aptos-move/framework/move-stdlib/sources/cmp.move

```diff
@@ -0,0 +1,98 @@
+module std::cmp {
+    const EQUAL: u8 = 1;
+    const LESS_THAN: u8 = 0;
+    const GREATER_THAN: u8 = 2;
+
+    enum Ordering has copy, drop {
+        /// First value is less than the second value.
+        LessThan,
+        /// First value is equal to the second value.
+        Equal,
+        /// First value is greater than the second value.
+        GreaterThan,
+    }
+
+    public native fun compare<T>(first: &T, second: &T): Ordering;
+
+    public fun is_equal(self: &Ordering): bool {
+        self is Ordering::Equal
+    }
+
+    public fun is_less_than(self: &Ordering): bool {
+        self is Ordering::LessThan
+    }
+
+    public fun is_less_or_equal(self: &Ordering): bool {
+        !(self is Ordering::GreaterThan)
+    }
+
+    public fun is_greater_than(self: &Ordering): bool {
+        self is Ordering::GreaterThan
+    }
+
+    public fun is__greater_or_equal(self: &Ordering): bool {
+        !(self is Ordering::LessThan)
+    }
+
+    #[test_only]
+    struct SomeStruct has drop {
+        field_1: u64,
+        field_2: u64,
+    }
+
+    #[test_only]
+    enum SomeEnum has drop {
+        V1 { field_1: u64 },
```

Comment:
For good test coverage, a nested enum might be helpful, and have a field with a vector. Also strings and signers would be interesting.